sympy/functions/special/tests/test_singularity_functions.py

```diff
@@ -46,7 +46,7 @@ def test_eval():
     assert SingularityFunction(5, 6, -2) == 0
     assert SingularityFunction(4, 2, -2) == 0
     assert SingularityFunction(4, 4, -2) == oo
-    assert float(SingularityFunction(6.1, 4, 5)) == 40.84100999999997
+    assert (SingularityFunction(6.1, 4, 5)).evalf(5) == 40.84100
```

Comment:
This is still not correct. The right side is a python float where as the left side is a SymPy Float with 5 digits. You could change the right side to `Float('40.84100')` to fix it. It is incorrect to compare two Python floats using `==`.